cpp/dolfinx/fem/CoordinateElement.cpp

```diff
@@ -74,42 +112,54 @@ void CoordinateElement::compute_reference_geometry(
     return;
 
   // in-argument checks
-  assert(x.cols() == this->geometric_dimension());
-  assert(cell_geometry.cols() == this->geometric_dimension());
+  const int tdim = this->topological_dimension();
+  const int gdim = this->geometric_dimension();
+  assert(x.cols() == gdim);
+  assert(cell_geometry.cols() == gdim);
 
   // In/out size checks
   assert(X.rows() == num_points);
-  assert(X.cols() == this->topological_dimension());
+  assert(X.cols() == tdim);
   assert(J.dimension(0) == num_points);
-  assert(J.dimension(1) == this->geometric_dimension());
-  assert(J.dimension(2) == this->topological_dimension());
+  assert(J.dimension(1) == gdim);
+  assert(J.dimension(2) == tdim);
   assert(detJ.rows() == num_points);
   assert(K.dimension(0) == num_points);
-  assert(K.dimension(1) == this->topological_dimension());
-  assert(K.dimension(2) == this->geometric_dimension());
+  assert(K.dimension(1) == tdim);
+  assert(K.dimension(2) == gdim);
+
+  // FIXME: Array and matrix rows/cols transpose etc all very tortuous
+  // FIXME: tidy up and sort out
 
   const int d = cell_geometry.rows();
   Eigen::VectorXd phi(d);
   Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> dphi(
-      d, _tdim);
+      d, tdim);
+
+  std::vector<Eigen::ArrayXXd> tabulated_data;
 
   if (_is_affine)
   {
     Eigen::Matrix<double, Eigen::Dynamic, 1, Eigen::ColMajor, 3, 1> x0(_gdim);
-    Eigen::Matrix<double, Eigen::Dynamic, 1, Eigen::ColMajor, 3, 1> X0(_tdim);
-    X0.setZero();
+    Eigen::ArrayXXd X0 = Eigen::ArrayXXd::Zero(1, tdim);
+
+    tabulated_data = basix::tabulate(_basix_element_handle, 1, X0);
```

Comment:
Shouldn't this be 0?

Reply:
The docs for this function can be found at: https://github.com/FEniCS/basix/blob/main/cpp/finite-element.h#L178-L189
I guess we need both 0th and 1st order derivatives to compute the jacobian, so I guess it is correct as is,

Reply:
I've requested documentation of this in FEniCS/basix#98